src/lib/vocab/community/communityServices.ts

```diff
@@ -234,6 +243,11 @@ export const LeaveCommunityService: ServiceByName['LeaveCommunity'] = {
 			// 	repos.persona.findById(persona_id),
 			// 	repos.community.findById(community_id),
 			// ]);
+
+			if (actor !== persona_id || !(await checkCommunityAccess(actor, community_id, repos))) {
```

Comment:
Two bugs here:

- `checkCommunityAccess` needs to be unwrapped, this would always evaluate to false
- `checkCommunityAccess` doesn't seem right? should be `isPersonaAdmin` right?

```suggestion
			if (actor !== persona_id && !(await isPersonaAdmin(actor, repos))) {
```